ProcessLib/BoundaryCondition/BoundaryCondition.cpp

```diff
@@ -90,46 +85,11 @@ BoundaryConditionBuilder::createDirichletBoundaryCondition(
     const unsigned /*shapefunction_order*/,
     const std::vector<std::unique_ptr<ProcessLib::ParameterBase>>& parameters)
 {
-    std::unique_ptr<MeshGeoToolsLib::SearchLength> search_length_algorithm =
-        MeshGeoToolsLib::createSearchLengthAlgorithm(config.config, mesh);
-
-    MeshGeoToolsLib::MeshNodeSearcher const& mesh_node_searcher =
-        MeshGeoToolsLib::MeshNodeSearcher::getMeshNodeSearcher(
-            mesh, std::move(search_length_algorithm));
-
-    // Find nodes' ids on the given mesh on which this boundary condition
-    // is defined.
-    std::vector<std::size_t> ids =
-        mesh_node_searcher.getMeshNodeIDs(config.geometry);
-    // Filter out ids, which are not part of mesh subsets corresponding to
-    // the variable_id and component_id.
-
-    // Sorted ids of all mesh_subsets.
-    std::vector<std::size_t> sorted_nodes_ids;
-
-    auto const& mesh_subset =
-        dof_table.getMeshSubset(variable_id, *config.component_id);
-    auto const& nodes = mesh_subset.getNodes();
-    sorted_nodes_ids.reserve(sorted_nodes_ids.size() + nodes.size());
-    std::transform(std::begin(nodes), std::end(nodes),
-                   std::back_inserter(sorted_nodes_ids),
-                   [](MeshLib::Node* const n) { return n->getID(); });
-    std::sort(std::begin(sorted_nodes_ids), std::end(sorted_nodes_ids));
-
-    auto ids_new_end_iterator = std::end(ids);
-    ids_new_end_iterator = std::remove_if(
-        std::begin(ids), ids_new_end_iterator,
-        [&sorted_nodes_ids](std::size_t const node_id) {
-            return !std::binary_search(std::begin(sorted_nodes_ids),
-                                       std::end(sorted_nodes_ids), node_id);
-        });
-    ids.erase(ids_new_end_iterator, std::end(ids));
-
-    DBUG("Found %d nodes for Dirichlet BCs for the variable %d and component %d",
-         ids.size(), variable_id, *config.component_id);
-
+    // TODO (naumov) Remove parts of the bc-mesh on which the variable is not
+    // defined. This used to be a filter on found node ids running over the
+    // bulk-mesh subsets.
```

Comment:
Why should a variable be undefined on parts of the boundary? Due to different ansatz functions (e.g. HM)?

Reply:
Exactly. And also for fractures I think.
Interestingly all the ctests are running anyway. Maybe the filter is already in the DOF::deriveConstrainedBC()... I'll check again, and possibly remove the comment.